index.js

```diff
@@ -601,6 +601,17 @@ class Analytics {
     }
 
     if (!this.queue.length) {
+      if (this.pendingFlush) {
+        this.logger.debug('queue is empty, but a flush already exists');
+        // We attach the callback to the end of the chain to support a caller calling `flush()` multiple times when the queue is empty.
+        this.pendingFlush = this.pendingFlush
+          .then(() => {
+            callback();
+            return Promise.resolve();
+          });
+        return this.pendingFlush;
```

Comment:
Add a catch block.

Reply:
How do you expect errors to be handled in this case? Do we just propagate the error again to all other callbacks?
The pending promise already contains a catch block with all the proper error handling, which will call the "original" callback if needed.

This is the scenario I have in mind:
- There is a pending flush already, with a callback from whatever code called `flush()` to create the pending flush.
- Some other consumer of the lib wants to flush as well, but no new events were added to the queue (this is what triggers the code in this PR).
- If the pending flush fails and we propagate the error, both the first callback and the callback from the other consumer will get the error.

I'll defer to what you think is best for the library to do, but I'm leaning towards resolving the callback without an error (just a `callback()` call) and throw the error again to keep it going through the chain. This "makes sense" to me because the `flush()` call on an empty queue wasn't the one that had the error, so in that sense we're maintaining the assumption that `flush()` on an empty queue will just call `callback()` without ceremonies.

Reply:
Make sense.